### ISSUE TITLE
Map inventories to Glottolog dialects

### ISSUE THREAD

**Issue by xrotwang** [phoible/dev#147]
It would be useful - I guess - to be as specific as possible when mapping inventories (or the corresponding varieties) to Glottolog languoids; in particular, for cases where PHOIBLE has multiple inventories from the same source for the same Glottolog language, linking to dialects would be good.

In some cases this also seems pretty straightforward, e.g. for the [language Giryama](http://phoible.org/languages/nyf) PHOIBLE lists 6 inventories, each identified with a `LanguageName` in the source that seems to map to one of the dialects of [Giryama in Glottolog](http://glottolog.org/resource/languoid/id/giry1241).

**Comment by xrotwang** [comment]
Another case is [Sebat Bet Gurage](http://phoible.org/languages/sgw), where 5 out of 6 inventories seem to correspond to [varieties on Glottolog](http://glottolog.org/resource/languoid/id/seba1251).

**Comment by drammock** [comment]
agreed. can this be (semi-)automated?

**Comment by bambooforest** [comment]
I started working on this issue here:

https://github.com/phoible/dev/compare/master...bambooforest:issue147

but I see that:

1. there's going to be a lot of these one-to-many mappings from ISO-to-Glottolog code
2. they aren't all so straight forward, e.g.

1465	ior	inor1238	Ennemor	gm
1466	ior	inor1238	Endegen	gm
1467	ior	inor1238	Ener	gm

http://glottolog.org/resource/languoid/id/inor1238

dialects:

Enegegny
Gyeto

the former which seems orthographically in between Ennemor and Endegen. haha.

Not that Wikipedia is better than the source dialects, but it lists:

https://en.wikipedia.org/wiki/Inor_language

"Endegegn, Enner, Gyeto, and the extinct dialect Mesmes are all sometimes considered dialects of Inor."

and the source documents describes Gumer and Ennemor, but also mentions Chaha, Soddo, and more.

Soddo is under [gru]

http://glottolog.org/resource/languoid/id/kist1241

and lists Soddo and Dobi whereas we have:

1456	gru	kist1241	Soddo	gm
1457	gru	kist1241	Goggot	gm

So regarding 1, do you have any magic method already to help out, @xrotwang ?

Regarding 2), this will take some detailed reading and consulting with Harald to match codes or update Glottolog.

**Comment by xrotwang** [comment]
by 1) you mean: "Multiple inventories with the same ISO and glottocode, but different language names" and you are looking for a way to get more fine-grained glottocodes based on these names?

**Comment by bambooforest** [comment]
yes -- because although we have mappings like this:

2864	nys	nyun1247	Balardung	er
2865	nys	nyun1247	Bibbulman	er
2866	nys	nyun1247	Goreng	er
2867	nys	kani1276	Kaniyang	er
2868	nys	nyun1247	Minang	er
2869	nys	nyun1247	Pinjarup	er
2870	nys	nyun1247	Wardandi	er
2871	nys	nyun1247	Wajuk	er
2872	nys	nyun1247	Wiilman	er
2873	nys	nyun1247	Wudjari	er
2874	nys	nyun1247	Yuwat	er

the "language" names line up to entries to Glottolog with distinct codes:

Bibbulman
Goreng
Kaniyang
Pinjarup
Wajuk
Wardandi

http://glottolog.org/resource/languoid/id/nyun1247

(of course it's also the case that they don't match spellings, as in Saraiki != Siraiki ... my any mileage here for automation would help)

**Comment by xrotwang** [comment]
I don't have packaged code for this right away, but it shouldn't be too hard to cobble together using the pyglottolog API and a library like [fuzzywuzzy](https://github.com/seatgeek/fuzzywuzzy).
Something along the lines of
```python
def matching_dialect(glottocode, name):
    lang = glottolog.languoid(glottocode)
    if fuzz.ratio(lang.name, name) < 95:
        ratios = []
        for dialect in lang.children:
            ratios.append((dialect.id, dialect.name, fuzz.ratio(dialect.name, name)))
        if ratios and max(r[2] for r in ratios) >= 95:
            return sorted(ratios, key=lambda r: r[2], reverse=True)[0]            
```

**Comment by xrotwang** [comment]
Yeah, this should work:
```python
from fuzzywuzzy import fuzz


def matching_dialect(glottolog, glottocode, name):
    lang = glottolog.languoid(glottocode)
    if fuzz.ratio(lang.name, name) < 95:
        ratios = []
        for dialect in lang.children:
            ratios.append((dialect.id, dialect.name, fuzz.ratio(dialect.name, name)))
        if ratios and max(r[2] for r in ratios) >= 95:
            return sorted(ratios, key=lambda r: r[2], reverse=True)[0]


if __name__ == "__main__":
    import sys
    from pyglottolog.api import Glottolog

    print matching_dialect(Glottolog(), *sys.argv[1:])
```
results in:
```
$ python match.py nyun1247 Bibbulman
(u'bibb1234', u'Bibbulman', 100)
$ python match.py nyun1247 Balardung
None
```

**Comment by bambooforest** [comment]
fixed in PR #147
